plugins/communications-app/RecipientsForm/index.jsx

```diff
@@ -0,0 +1,187 @@
+            description={recipientsFormDescription}
+          />
+        </Form.Checkbox>
+        {
+          // additional modes
+          hasCourseModes
+          && courseModes.map((courseMode) => (
+            <Form.Checkbox
+              key={`track:${courseMode.slug}`}
+              value={`track:${courseMode.slug}`}
+              disabled={hasAllLearnersSelected}
+              className="col col-lg-4 col-sm-6 col-12"
+            >
+              <FormattedMessage
+                id="bulk.email.form.mode.label"
+                defaultMessage="Learners in the {courseModeName} Track"
+                values={{ courseModeName: courseMode.name }}
+              />
+            </Form.Checkbox>
+          ))
+        }
+        {
+            // additional cohorts
+            additionalCohorts
+            && additionalCohorts.map((cohort) => (
+              <Form.Checkbox
+                key={cohort}
+                value={`cohort:${cohort}`}
+                disabled={hasAllLearnersSelected}
+                className="col col-lg-4 col-sm-6 col-12"
+              >
+                <FormattedMessage
+                  id="bulk.email.form.cohort.label"
+                  defaultMessage="Cohort: {cohort}"
+                  values={{ cohort }}
+                />
+              </Form.Checkbox>
+            ))
+          }
+        <Form.Checkbox
+          key="track:audit"
+          value="track:audit"
+          disabled={hasAllLearnersSelected}
+          className="col col-lg-4 col-sm-6 col-12"
+        >
+          <FormattedMessage
+            id="bulk.email.form.recipients.audit"
+            defaultMessage="Learners in the audit track"
+            description={recipientsFormDescription}
+          />
+        </Form.Checkbox>
+        <Form.Checkbox
+          key="learners"
+          value="learners"
+          className="col col-lg-4 col-sm-6 col-12"
+        >
+          <FormattedMessage
+            id="bulk.email.form.recipients.learners"
+            defaultMessage="All Learners"
+            description={recipientsFormDescription}
+          />
+        </Form.Checkbox>
+      </Form.CheckboxSet>
+      { isFormSubmitted && selectedGroups.length === 0 && (
+        <Form.Control.Feedback
+          className="px-3"
+          type="invalid"
+          hasIcon
+        >
+          <FormattedMessage
+            id="bulk.email.form.recipients.error"
+            defaultMessage="At least one recipient is required"
+            description="An Error message located under the recipients list. Visible only on failure"
+          />
+        </Form.Control.Feedback>
+      )}
+    </Form.Group>
+  );
+};
+
+RecipientsForm.defaultProps = {
+  cohorts: [],
+  courseModes: [],
+};
+
+RecipientsForm.propTypes = {
+  cohorts: PropTypes.arrayOf(PropTypes.string),
+  courseModes: PropTypes.arrayOf(
+    PropTypes.shape({
+      slug: PropTypes.string.isRequired,
+      name: PropTypes.string.isRequired,
+    }),
+  ),
```

Comment:
In the current implementation the `courseModes` array is required

https://github.com/openedx/frontend-app-communications/blob/6dd835d63f106c6e1bf7a02de16d3047a5bee558/src/components/bulk-email-tool/bulk-email-form/bulk-email-recipient/BulkEmailRecipient.jsx#L130-L135

Is there a reason it was switched over to defaulting to an empty array instead?

Reply:
`courseModes` sometimes was undefined even if the prop is required so I think is better to pass a default value for that prop and remove the required